### ISSUE TITLE
Svelte 5: Not removing 'data-is-dnd-shadow-item-internal="true"' property so elements get hidden permanently

### ISSUE THREAD

**Issue by vini-brito** [isaacHagoel/svelte-dnd-action#596]
Hello! In Svelte 5 I am having an issue where once I release an element, it keeps the HTML property added by the library `data-is-dnd-shadow-item-internal="true"`, which means the "visibility=hidden" CSS property is kept, which means the element is still there, still occupies space but cannot be seen, loses visualization.

In Svelte 4 it works as expected.

Using the migrate script doesn't help.

In the reproducible example below, just drag the elements around and they will lose their visualization but will still exist and can be inspected via regular context click -> Inspect to see that the HTML property is still there. 

[Reproducible in Svelte 5 preview here](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAA-0aC1Mbx_mvfFESI1y9MMWND44MQYrNGAcqcFOP5WFWt99JO5x2r3t7gKzov3f2pXsgUeO2004CA0K3-71fe_d9t2jELMGsEXxcNDiZYSNoHKVpo9VQ81RfZDeYKGy0GpnIZaRXDrJIslQdjvhIsVkqpIKF_j5SlNPPgmPLXl28Oeqf_Xp1fnp0PHhzdtofDK9OLgfvrk76DuByePL69WB4UYU3MO-Ohm8Hw6vz4dn5YHj54eqXo3cDD3b2fng8uNAXS4ilmMGWlbFNOW2TSDHBt_bLwkGcsBSqwF3C2YwoNJBl2EgiUTi4Qa76LEuJiqYoa8geqYJ4K5ki4wTrjDIlpMEAAOh2oY8x4whqimC2MqBMYqSSOTAOasoyiMQsFRy5skh4Z1hEgmcKMkwwUkhPFM5OKIQrvk2eJ8n2_hoULihmx5pmhhWMj5-2rSZ8pCwodSpDuN4QTYfgwLVh-7kk2ubvMghhr-f2E1QwEzKd9lmmeWm-SuZY2o6QK4l9SSYTpGf8OJeZkBBCTJKsDEclmZSolLfjnBt_Q6aIVJpUk9EW4LaPSOykEo0KGJM8UVZ-HaobiKqRqlq4k6FqMmrxljW2U8Jpgm9x3he3vFmwZTE0m9i5xjmEYQhbA65QbsFvv0FpEba24dmzinrbG5X1rJ1jtV2Ywpm2-Ufr8oX9p39GjYgonAg5HzUCGDXGiYiuR41WBUJwpSPMAFzinQ2TGpAiEwuQ1jYYtesK79TV3vXP6mWkzpLT_Oru7-Pjo58-D0__ehfVkVIpUpSKYaaRS_LabS5mfTE7rwB9rALV9KzhkxlaqU7JGJMK8ypkRFIyZglTzj4DytQD4NTGzt9IkmNhr1HjPvxyDY0vkPfY--K_KLGuKywDAuo_K_s5kcjVCX2E8EMkD4HfF36tuNWlT616ONEviaU6leLS7Xir_O-y6x9_eZWf9nZ3CWV8Hr_7NY0n_dmOfP3qKbuesut3kF36vPy_ya7d3af8eqy8T_n1sPB_9Pwi5CnDvFueMuwLhH_KMPiqE2xv7ynDnjLsS4R_yjB4bIYRQsZPpxjAU479HnNsxD_t19qLuv9fX6OYKt0b7tU3MjQNUdOQ3pJCqAssGvGujzsl2fGUJVQir7Vxvwtqm7a1-eyZ_dJJkE_UFA6hV-_72gZsn9NjwTNGUZaasN0uHEURZpnptVNUhCXgsnMOIjbLpjls4W-I9FAhYMd-9Z3-GjUrIOEUGI8FFDnv6VrsgjDH2xPXr7VbHUNiv4AwhIpdHosNvO0gxjBXkk0mKNfw1_gFbYcSmuWOvSqx9mTcvrtcsfeNZq9DsRFD05MOQ3CTmc7bwYefzo6Gfe3AFekwBD_v6fSHR6-vLi7Pzs8H_ZW77vXk_bhA2eb3huZ7n9OfGScJ-4x_HN_XNfkybz_szHE-Hif4Pj2TF6guJeJq1rPR0ednJ79cDob_rgu1HMcJi67t8EZ7pRjg6KtOpkSqyxyZmFFTc3tfT9LO7WjHYtihm1GC8QnkKSgBqSnbgAnOkKusUOa7-hgtDIHRkiLdLpxY20ZaMqTG7eZsSSQSOl8N4jzGmrGRnchpUd9zu70KIGcUwCTDguuG0ZOmcIEW2YM4eSjkmdZ3NU58wNprPFweWLniHobQKxmiMjy0Eumw8XO0ugarQHAjw-ZWnlKijI-zre1aMBx0i1kyP3CHiJmq0nCxOlMMdJSQLAv1uZmq6cJ8LvWJ6XYCLvz5ES6-KZ0mBjnPMHDD6XCxsBncqk0vW-bEa1Wnlq0NU8qlISt45I6dcHHvJHIQsStOJQhfr5Yjboboi2-RRFN31sUsUSibxsrbEB6a5Q6j8I1OvI0j9W0gmQ2JpkPYXlpbH1B2473iht6B1lwbgjrtg_okd7n0KN7ujN-wjI0TbGt6jcp24IMyXKxNLCvQiqTgJqfCRdPlenh4vxJYHF8OHO6hJ7H4lsWWrn8SsDNVfddPGEe5tSxicaQO7CxeyxkXqyM1ZpwGxu7hwnLU35dlEB0U4aixKiGN8qYJw9BGI_wJdiqYq_j1tGllW_BSZoSLe8lZAu4e-kJcd6iRkGg18C5cVKrvj9CDANpVmYhkpJ3oBwedSpJM2tbwVbW8z9ftZWqe4D1WW5HJiwAmkoy3IKgujBmfbNWUn4k8QypueSUMSnP8tQHgsZXIo6kB_ir0a5zfY70og4xUbbRvwParIBu5VeCWJd6HB13Kbooo7rJ4Wbh2U1Cbx9uy_Q5SYKWoApuEUZ4pMWtnU0LFbVtvWpCP_-rFmk_Lw7LAlq57fi5z7ab3JDfLXqVFV9cxW9Zt7Lu6rkPGQFB24-08RTaZqgBiptqOlzPbLaNqum7Dh1QqmH6fwq2mImO2hklMiGI3lfclKLvprI5Mx3osJEUZwIv0DjKRMArjJK9gOfG_ArNTqZMeb6NKm62QEkoZnwTuWWukZkROGC-uvSxc8KoEphr1POsZ423PZafX-36_WHZS7e710jtPlETXEylyTtuRSLSx5WRMmjsvd1qws_uD--h1env6ruQ5HEGWj1WCUCCCQYTn3aqYOyuTGYovdl-2YHfX_vU6O55coiUFi1SicefCOoBeemfM_-f0zlLqtcD9FlQyS8birMiIlERMzQPodV4ZwCHSPEJw6xALCcQrhHGs79c8rpKE-zjz8L3ObgZIMmwz3ha5sswp1URmQqgpFFiOUDW-Ai5U0_lru54YJFeimhGllcf4aVPoWoxXvRa8NL_Gfo82-BrTPsJiFlTfLCZkru9C8G5dHhY3d3UrvSpi11lpteJodJ_DG3GD0ntUCZiRa30Xrm_1ELhQLELz5p7gYCqLfYnQucz7amqILGr67mywmLbWD-sstld5k6xzv2h7FqaMmH5SAK6k7N9PKUqyKVKYSJzfC43AZtOqRFXctGui9VTcoly5pxKjHXsCrt7t9D4y9UbjvmHUvsboAMdzcE2tVdYUtZmMM5HkygsirffaLwr3KZEGsLcqUCZ8YiFngc2ihCj80Gzv9b7frvq7RMIHxYuHClrJHOV7lDW2rVV452T22ZRlC9geC8_pc9vchQXwonb-uMjZZE9zuhuDXkzFbdmgt1Pk5to9xuoDhWVgyCEtvHXQ9Sdso9WYCcpihrQR6Ifv5aflPwEIG9m-4SsAAA==)

**Comment by isaacHagoel** [comment]
Thanks. Most likely it's a svelte 5 bug and not a lib bug but I will block
some time to have a look

**Comment by vini-brito** [comment]
Thanks! If there is anything else I can do let me know, I can spend more time on it to help debug or even fix in case it is a low hanging fruit a non-maintainer could do. It is the only external dependency in my project, so it would be worth it for me to be acquained with its internals.

**Comment by isaacHagoel** [comment]
hi,
I had a quick look. If i remove the filtering of the shadow item on line 285 (#each items...) the problem seem to disappear. This example is quite complex and it is hard for me to understand what you are trying to do just from looking at the code. Can you provide some context?

**Comment by isaacHagoel** [comment]
also I pasted it to a normal REPL (not Svelte 5) and it is broken as well. I think the issue is in the code. If you explain what it is supposed to do I can try to help you fix it

**Comment by vini-brito** [comment]
Ah it looks a bit odd because I stripped some code down from my actual page contanining this, brought stores into it etc.
For context, it is a drag and drop webpage builder, from the viewpoint of the library it is a recursively nested element, using `<self>`. 
I had some ideas and I will go through them sometime this week in order to debug and I will let you know.
I appreciate you looking into it!

**Comment by vini-brito** [comment]
Ok, so I properly rewrote it to Svelte 5 and inspected things some more.
A difference that I found is that in Svelte 4, items are always objects.
Now, on Svelte 5, props are not regular objects but rather proxy objects:

![image](https://github.com/user-attachments/assets/723c70b1-0c03-4e7a-8680-92447deb7aa2)

I bet that interferes with the library functions, when it tries to target regular objects and finds these proxies.
Apart of that, I did not find anything else different from 4 to 5 in the data that is passed around. 
Not saying that this proxy thing is the problem, just that it is the only difference I could see, of course it could be something else.

Here is the [properly updated reproducible example](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAA81YC28bNxL-K1M1jaSetJLtc9CuvT44li8x2tSG7NyhiAKXWo4kXlbkguRaVlT99wNfq109cm2BQwvbspacF-fxzXBXjQnLUDXiD6sGJ3NsxI3LPG90GnqZmwf1hJnGRqehRCFTszLS5yqVLNcXI87muZAaXhcso2ScIUykmEMz6pUrkRPQPBvxER_pkZ4UPNVMcJgiR0k0DgmnYn5DWxnyqZ61YeUIU8GVBiIlJMBxAe8Z199dSkmWnhJ6cNw-g14P3nhRMCPZBPQMwVOMMSWFQkCSzmC81AhT9oQK9ELADJ-BsinTyulbME7FIkrlMtcimqJ2hv2LZAWqFpGyfeYIJepCcrCmRObAZrMDFFNILsy_SIt7LRmfto5etaOc0HtNpG4dd2DU6I8a7Xb0H8F4q9l0EtfGNRlq4IKiuhLzXCikkMAH74mRTonGqZDLGJrjTKSfmp1yR3CNXMfQfMBnJ2Kzqck0hma-WWA0hl80PuvHF6sd_x-ftte_eNJ1yVL_-ZMs-iva5LT_RS1b7zeo8ydYFCiDZR_N__NeiSHnG_RgGucqWdXqYA0GiZJRAzOcI9dq1IDeRaPTmAvKJgxpI9aywHWnxK9S3pdQbINhI-1RzPuGcvpZcPTm3r-9HNz--_Hux8ur67e3Pw6uh483D9fvHm8GnuBhePPmzfXwvk5vad5dDn-4Hj7eDW_vrocPPz_-dPnuOpDdvh9eXd_b8veg6WzsUk67xCKkxczSOJhkLIc6cY9wNiceXbWrk5J-IZm2Xt3iUVpIywEABjwHOGEcLWraLQWUSUx1tgTGQc-YgtTEgiPXhgmfrQIHzwozTDXSG43zG4NZQWuLF1nWPtth2Ma4kv7Dx7bvEY7QHHdQSGI88U5BAqd9v2-gci5kPhswZViNGJMCle0UuZY4kGQ6RXrLrwqphOkjE5KpKh2VZFqR8kJporFlqYI1Zb9SBsSNyBajHcCyT2GUS3xCrgc4IUWmW6FPmHOIDKNMTFujhmW3CoGpUcNI8PlWt6E0UY903buRQt1itNI0KtbNCKcZ_oDLgVjw1sY6NoFWC6NPuIQkSaB5zTXKJvz6K1QWodmGly9r3mhv-2Zj13rjv5UrWeO6MeM0BLJja9a0xFzPIIF-BxRaO22ONKUQ-t4tNDuQE4lcvy7G4wzf57fyHvWDRJO3CbzIpchVq31W1Tkj6mrGMiqRB8McMUXJnpC2nFEvXzrrIj8MXEB_J6jObQNOrwRXjKKsuK7Xg8s0RaVsbVDUhGVg7EGplyDcoGEj7-ifiAxUCWDkvu7NhdQrq2bCHpXuGIRTYHwigmqGKih3KjbaOS5ufDzcVmRFnG0orKDNLp-Is_26HVBa5Vqy6RTlHv2GfyPbsyR2OXJPFdVBjN_3j6X6kEjhDJuNCbSC6CQBj5zRD9c_v769HA5MmEvRSQIBj6PB8PLN4_3D7d3d9aCM6U65BeCo9Ex-uL4GnP6TcZKxz_j_SJS_Zg5sn-Q3R72e9BPvuJ2k3xf5crQab8NCiQUHM-Pu9uanh-vhb4_5_lgbW64yln5yeG_Ct8F88xQpLfI7KXIytV2q5S4jd64bOA7Xeu0hGJ9CkYMWHvCgHGjKw7zYbqdJAoxWDtLrwY0LQmosQ2rzA5gCkkkkdFk25MCxp4W4zmxMfc_ddplp3imABlJXX5Bh2pCRcI-OOZB4eygUypy3HCq-4O09Ea42L99DkgT6FUfsN_RAJyk7ctBfmT5H_Ny3Jjs80WRVdipLnWZEqWTUsGas7Od61Ch3Yi5CK0pWX1Uak2UuFMZ-nExWvld2tiab0CtrE03nwASztmIFDw0kWe00ME8Rqq1CEZBrPeJ27F19ba_FrkdOWKZRtmzrbJt7rPkSMQpfmco6OAS3gSgX85ZnaHsYPafsKcTFj6mxOblxBPWnj7envHV5bwl-Z_yJKTbOsGvkNWrbcci6ZLW3cpxBpUjBbdEkq5Yv5uRit9QdT6h3z3sRRKy-ZhMnN9yf3ABl7kyEcZTNysXr3A3cxspJ9TpmJqXYX3WcPvO9dmPbufJUN20SJi4X4W9wVOMsszfIprXtAxWSrHaqsMLWuwiIux1YdyU0B8LnZFWD2X9AH2Lo1q0jkpFuRsaYmZKSZNp1AagfMMR-357Sywx3VDVTWx8xTCUZNyGuL4wZnzZrdgg-F4VCKha8lg6VGX9vIgRuLYp0Zon_EPsnXO6oXlVJRnprnrdkZ3WSg9pqdOuK7ovzHmVPm2zuscl6E9pDye3eD1QTOwdWyS9wxZgWSot5V80IFYuu2XQkH_7Xlfjj-qJq8A2QORBw4v3LiKryXr5zALscTrbqGVhzKO-KwcO8yRxLQdlTcPcM2XSmY5gw3fW6vPcWjOrZvo2QWblg5i7lV3OhmIM0iRnR7Kl2V6LsKSpbpFc9FpKijOE4fwYlMkZhnBU1Lm_-H-CMarAZ-A4e6bAXckIp49MY-n5hTuSU8c1zsIULXrfAwlM_qJ4z3g1ajvr9b842y96qk9N-_hyEkvTTVIqC024qMuNsOR2T1tGrow4cnXznP_pR_9RMId_CJahirDOEDSNYRvi2VzfzqHSZlXh88qoDJyfurx8dBXGZsRQcU0XGs8_uGPr5s3X_3_NnJ6nfAf-7kaKcGMdTihE5SZlextCPvreEQ6RFiuDXYSIkkHAgnEzMfBZ4tSQ85Fmg70cnCpAo7DLeFYV2yik1QuZC6BlsuLygen7FXOiWj1d7uzBIoUW9IiorvydOh1LXcXzf78Ar-2v997sdvse1v8NjjpQylWdkaYYSfN5Xh5tZb9tL329y13upXPEyet_CW_GEMkRUC5iTT2bqNpMfAheapWjf1wkOFlncW0AfshCrmRWy2jrv0QGPGW99t89jp7W3SNEudgcVFkZYZrV4SDnbLSlK1AwpTCUud1IjdtVUQlQtTCc2W38UC5RleGo5GrlGWL6cDTGyeGN43zLqXl56wvESqHsRV1bNBpvJWIms0MEQ6aLXPd6ET4s8htMSoGz6TIScx66KMqLx51b3tP9Nux7vioiQFMdfArSKO6qjyh7fbiG8DzL7bGHZEXbHImj63LXDWAzHW_3HZ84hf9ombx16PxOLqkMXM-T22V9bTUNhCqw4pJtonfdCh919Of9x_V8CJtxJaBwAAA==).

And removing the filtering of the shadow items indeed solves that problem, but breaks the nested implementation, this is how it looks broken:

![image](https://github.com/user-attachments/assets/8effdd12-95c2-4d8e-bee0-860d7116d9d2)

And how the Svelte 4 version looks, which is the functioning version:

![image](https://github.com/user-attachments/assets/1a652ece-a064-4b8f-923d-d0fe455ad070)

The filter idea I got from the docs: "_you might want to import SHADOW_PLACEHOLDER_ITEM_ID in order to filter the placeholder out when passing the items in to the nested component_" just to give context as to why I wanted to have the filter too.

A Svelte 5 version showing how the [container breaks is here](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAA-1YCW8bxxX-K6-MY5IpuaSkykhWWhWypDpC6kiQ5BaBaTjDnUdy6uXMYmaWFM3wvxdz7cEjjQMUKIoAsqydeff1vd11a8IyVK34_brFyRxbcesyz1u9ll7l5kEtMNPY6rWUKGRqTs5VKlmuL0Z8pNk8F1LD64JllIwzhIkUc2hHg_IkcgLaZyNuGCYFTzUTHKbIURKND4RTMb-lnQz5VM-6sDZkI50KrjQQKSEBjkt4x7j-9lJKsvKUMIDj7hkMBvDGi4IZySagZwieYowpKRQCknQG45VGmLIFKtBLATN8BsqmTCunb8k4Fcsolatci2iK2hn2D5IVqDpEyu6ZI5SoC8nBmhIZd81lDzoU0y4kF0AxjbR41JLxaefoVTfKCX3UROrOcQ_aw3a3G_1LMN5pt53EjQtMhhq4oKiuxDwXCikk8N5p9BEZ6ZRonAq5iqE9zkT6qd0rbwTXyHUM7Sd8doKqS02mMbTz6oDRGH7W-Kw_vljv5OH4tLv52VFuen8Y8IcBhw0wGgnjKPcZcbV76YygbFEzQ-NcxfD-Q9OwUvJB63p_mPf_b57578PZiJ8PKszh5xXaWAXJujE3N2CQKxm1MMM5cq1GLRhcjHir15oLyiYMaSvWssBNr0S8UuIX4p6PG-X0s-AYvLS_nx5u37y5eXj0h4_fX17f_fPj7dPN249vLx9-uHn4eP9wd3_z8PTTxx8v394Esrt3D1c3jxYXPJY6Q_qU0z6x0GmhtLQAJhnLoUk8IJzNiQdd7fClpF9Kpm3wtniUFtJyAIBB1WucMI4WTu2VAsokpjpbAeOgZ0xBakLOkWvDhM9WgcNthRmmGumtxvmtgbKgtcOLLOue7TBsQ19J__5D168OjtC4e11IYiLxVkECp8OzCkHnQuaza6YMqxFj8ly7TpFrideSTKdI7_hVIZUwC8aEZKpORyWZ1qS8UJpo7Fiq7vYiowy2G5EdRnuA5QKDUS5xgVxf44QUme6EBcL4ITKMMjHtjFqW3SoEpkYtI8EXVdOG0kQ90s3oRgp1h9HGNlFaNyOcZvgDrq7Fkncq69gEOh2MPuEKkiSB9g3XKNvwyy9QO4R2F16-bESjux2byq7aIrN2nWlCN2achkT2bGv2gGKuZ5DAsAcKrZ22RtpSCP3oDto9yIlErl8X43GG7_I7-Yj6SaKp2wRe5FLkqtM9q-ucEXU1YxmVyINhjpiiZAukHWfUy5fOushviRcw3EmqC9s1p1eCK0ZR1kI3GMBlmqJStjcoasIyMPag1CsQbgO1mXf0CyIDVQIYuT_31kLqldUrYY9K5wbhFBifiKCaoQrKnYpKO8flrc-Hu4qsiLOKwgqqbvlEnO3X7aahVa4lm05R7tFv-CvZniWxx5F7qqkOYvy9fyzVh0IKPlQXE-gE0UkCfnJGP9z89Pru8uHapLkUnSQQ5nF0_XD55uPj0939_c11mdOddguDI6BQGKP7--ua078xTjL2Gf8bhfK_WQPbnvzmrDeLfuIDt1P0-zJfwut4eyyUs-BgZdzf3f74dPPw23O-P9fGlquMpZ_cvDfpq2a-eYqUFvm9FDmZWpTquLfUe4cGjsNBr3WC8SkUOWjhBx6Ue0vpzIttOE0SYLTmyGAAty4JqbEMqa0PYApIJpHQVQnIgWMPhDhkNqa-4-66rDQfFEAzUte_IsPAkJHwiI45kHh7KBTK-FsuFb8S7T0ZroOXx5AkgWEtEPsNPYAkJSIH_c0100OTXZ5osi6RylKnGVEqGbWsGWv7ezNqlTcxFwGKkvWfasBkmQuFsd8Zk7XHyt7WZhOwsrHR9A5sMBsrVvAAIMl6B8A8Rei2GkWYXJsRt7vt-iv7vcRPGOVSZ7EzYrTrp-E5ZYsQXr9txsYB4w_1TsTby9omjNIyfIwvmGLjDPtGXqtxHYfiSdZ7G8AZVIoU3NZ-su74nkwudjvW8YS29bwXQcT6KzZxcsN7kNuDqlehUttIn7u92Vg5qU5H2iw8sX8xcfrM35s6yc4LSv3S1lLiSgr-DEcNzrIIg2zauD5Q6Ml6p5lqbIOLMDi3E-te64xD-JysG9PyrzCEGPpN64hkpJ-RMWamMySZ9l0Cmg6G3O-7U3qV4Y6qdmrLPIapJOM2xM2DMePTdsMOweeiUEjFkjfKobaq7y2EwK1Fkc4s8e9i_4SrHdXrOslIb63lluysSXJQW4NuU9N9cT6gbFFV84BNNlVqDxW3-4pUL-wcWK2-wDVjWigt5n01I1Qs--bSkbz_T2-2HzYXdYNvgcyBgBPvPyjUlQ_yHQfscfBsPTDTyQ1r1wx-WpvKsRSULUK4Z8imMx3DhOm-1-Wjt2RUz_ZdhMrKBTOvRP40F4q5kSYxI5otGq88lC2iEum86rGQFGUMx_kzKJExCuOsaHB5838HZ9QYm4HvoEuHo5ATShmfxjD0B3Mip4xXz8EWLnjTAjuehkH1nPF-0HI0HH59Vh17q05Oh_lzEErST1MpCk77qchMsOV0TDpHr456cHTyrf81jIanZpn4Bi5BFWOdIVSMYBnhm0HTzKMyZFbi8cmrHpycuH_D6CiIy4yl4JhqMp59dccwzJ9t-P-SPztJwx74n0qKcmIcTylG5CRlehXDMPrOEj4gLVIEfw4TIYEEh3AyMWtW4NWS8FBngX4YnShAorDPeF8U2imn1AiZC6FnUHF5Qc36irnQHZ-v7nZjkEKLZkfUTr4kT4dK13F8N-zBK_tj4_fFAd8T2i-ImCOlTOUZWZmlBJ_39WG1sm1H6buqdn2UyhMvY_ANfC8WKENGtYA5-WSWZ7PAIXChWYr2s5vgYCeL-5jnUxZyNbNC1lv-Hh2ImInWt_sidtr4GBTtzu6gwo4RllktfqSc7bYUJWqGFKYSVzulEbtuKkdUI00ntlr_LpYoy_Q0ajRyQFh-SA05svPG8H7PqPsG6QnHK6Due1rZNdVsJmMlskIHQ6TLXv-4Sp8WeQyn5YCy5TMRch67LsqIxp86_dPh191mvmsiQlEc_9pAq4Wjvqrsie3WhPdJZp_tWHaE_bEImj737TIWw_EW_vjKORRPC_I2oI8zsawHdDlDbp_926cBFKbAikNaZet8EBB290P6h82_AZl013lGHgAA).

I imagine it breaks because the library needs me to filter, which is why you made the recommendation to filter the placeholder out and even implemented the ID of the shadow element, so proceeding with the "no filter" path unfortunately is not a solution for a nested implementation.

So! After all, I can try to debug these proxy things that the new `prop` feature creates, would you have any pointers for where I can start looking inside the lib? Or maybe things to try in my code, I mean, external to the lib, using it as it already is.

**Comment by vini-brito** [comment]
Just a heads up, I've been debugging this and I found out it has nothing to do with Svelte 5, I am debugging in Svelte 4. 
Maybe I will change the issue name if I can.

So far, what I am understanding is that when the decorator is applied the correct target element is not in the list, so the correct index ends up hitting another element next to it. 

Immediately after that another extra decoration happens and hits the correct element, which is the one being dragged.
However, later the undecorate action then hits only the correct target, leaving the incorrect one with still having the decorated style.

I will still dig further, find out why the correct target briefly disappears from the list, what removes it or more probable what applies the condition on it that will cause it to be filtered out.

**Comment by isaacHagoel** [comment]
Thanks for the update. Let me know if I can assist

**Comment by vini-brito** [comment]
I removed the filter and made several changes in userland lately and apparently this issue is gone, so the PR is unnecessary. However I cant isolate what change solved it, else I'd share it. 

Unsure if I should close this issue, while the filter issue is real, it is no longer an issue for me, so I'm leaving it to you Isaac whether this should be closed or if anything more should be done about this issue.

**Comment by dkmooers** [comment]
I ran into this bug in Svelte 4 when I had nested drag-and-drop (2 levels), where the state of the blocks was stored in a writable store (using Superforms > $formData):

```
$formData: {
  blocks: [
    {
      id: 'block-1',
      items: [
        { id: 'foo', productId: 'XYZ' },
        { id: 'bar', productId: 'ABC' }
      ]
    }
  ]
}
```

You can drag-and-drop blocks to reorder them, and within each block, you can drag-and-drop items to reorder them.

When I tried to drag-and-drop block.items inside the block, I was getting some buggy behavior where multiple shadow placeholder ID items would show up in the DOM during consider as I moved an item back and forth to different positions, and also if I only moved an item one slot to the right or left and then dropped it, the dropped item would still have the data-shadow-marker-internal attribute set (can't remember what it is exactly), and consequently the element would be invisible (visibility=hidden).

I fixed these bugs by just maintaining `blocks` as a local variable to the component, and using:

```
$: if (blocks) {
  $formData.blocks = blocks
}
```

to reactively sync the blocks state to $formData.

Not sure how to trace the origin of this bug, but the fix works for me!

**Comment by jdkdev** [comment]
Experiencing this as well.

**Comment by isaacHagoel** [comment]
I am happy to have a look but need a REPL for that

**Comment by jdkdev** [comment]
thanks @isaacHagoel !  

So, originally my items were a computed list:
```
$: dropdownItems = realList.filter(mySelectedFilters)
```

I worked around it by making `dropdownItems`  static.  So first question,  should it be able to work with a computed value?

My thought was that maybe because the elements are getting redrawn it's losing reference to the element that gets the `data-is-dnd-shadow-item-internal=true` and can't unset it.

**Comment by isaacHagoel** [comment]
Thanks for sharing.

**Comment by Den-MM** [comment]
> Thanks for sharing.
> […](#)

I'm sorry for posting here, but I found the issue. I was using an additional animation (fade) within the component that the each block loops over which probably disrupts the already used animation in the container of that component (dndzone).

```
<script>
     const flipDurationMs = 200
</script>

<div 
      use:dndzone={{items: yourItems, flipDurationMs)}} 
      onconsider={handleConsider}
      onfinalize={handleFinalize}
>
      {#each yourItems as yourItem (item.id)}
        <div animate:flip={{ duration: flipDurationMs }} aria-label={filter.label}>
          <Component /> <!-- inside that component should be no transitions/animations disrupting the dnd -->
        </div>
      {/each}
</div>
```

**Comment by isaacHagoel** [comment]
Closing in favor of #648 as the canonical current lifecycle/hidden-item report. Please retest with v0.9.77 first because releases 0.9.70 through 0.9.74 added directly relevant lifecycle fixes.
